src/app/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -8,27 +9,34 @@ import { CoursesService } from '../pages-block/services/courses.service';
   templateUrl: './breadcrumbs.component.html',
   styleUrls: ['./breadcrumbs.component.scss']
 })
-export class BreadcrumbsComponent implements DoCheck {
+export class BreadcrumbsComponent implements OnInit, OnDestroy {
 
-  courseId: string;
-  course: ICoursePage;
-  breadcrumbpsTitle: string = '';
-  defaultCourseData: ICoursePage;
 
-  constructor(private router: Router, private courseService: CoursesService) { }
+  course: ICoursePage;
+  breadcrumbsTitle: string = '';
+  private currentSubscribes: Subject<void> = new Subject<void>();
 
-  ngDoCheck(): void {
-    const regEx = /\d+/;
-    const url: string = this.router.url;
-    const res: RegExpMatchArray | null = url.match(regEx);
+  constructor(private courseService: CoursesService) { }
 
-    this.courseId = res ? res[0] : '';
+  ngOnInit(): void {
+    this.courseService.currentCourseId.pipe(
```

Comment:
remove this and send to subject course title immediately
do courseService as public and within template you will be able to use {{ courseService.currentCourseTitle$ | async }}